spring-cloud-task-batch/src/main/java/org/springframework/cloud/task/batch/handler/TaskJobLauncherCommandLineRunner.java

```diff
@@ -0,0 +1,195 @@
+import org.springframework.batch.core.configuration.JobRegistry;
+import org.springframework.batch.core.converter.DefaultJobParametersConverter;
+import org.springframework.batch.core.converter.JobParametersConverter;
+import org.springframework.batch.core.explore.JobExplorer;
+import org.springframework.batch.core.launch.JobLauncher;
+import org.springframework.batch.core.launch.JobParametersNotFoundException;
+import org.springframework.batch.core.launch.NoSuchJobException;
+import org.springframework.batch.core.repository.JobExecutionAlreadyRunningException;
+import org.springframework.batch.core.repository.JobInstanceAlreadyCompleteException;
+import org.springframework.batch.core.repository.JobRestartException;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.CommandLineRunner;
+import org.springframework.boot.autoconfigure.batch.JobExecutionEvent;
+import org.springframework.cloud.task.listener.TaskException;
+import org.springframework.context.ApplicationEventPublisher;
+import org.springframework.context.ApplicationEventPublisherAware;
+import org.springframework.core.Ordered;
+import org.springframework.util.PatternMatchUtils;
+import org.springframework.util.StringUtils;
+
+/**
+ * {@link CommandLineRunner} to {@link JobLauncher launch} Spring Batch jobs. Runs all
+ * jobs in the surrounding context by default and throw an exception upon the
+ * first job that returns an {@link ExitStatus} of FAILED.
+ * Can also be used to launch a specific job by providing a jobName. The
+ * TaskJobLaunchercommandLineRunner takes the place of the
+ * {@link org.springframework.boot.autoconfigure.batch.JobLauncherCommandLineRunner}
+ * when it is in use.
+ *
+ * @author Glenn Renfro
+ * @since 2.0.0
+ */
+public class TaskJobLauncherCommandLineRunner implements CommandLineRunner, Ordered, ApplicationEventPublisherAware{
+	/**
+	 * The default order for the command line runner.
+	 */
+	public static final int DEFAULT_ORDER = 0;
+
+	private static final Log logger = LogFactory
+			.getLog(TaskJobLauncherCommandLineRunner.class);
+
+	private JobParametersConverter converter = new DefaultJobParametersConverter();
+
+	private JobLauncher jobLauncher;
+
+	private JobRegistry jobRegistry;
+
+	private JobExplorer jobExplorer;
+
+	private String jobNames;
+
+	private Collection<Job> jobs = Collections.emptySet();
+
+	private int order = DEFAULT_ORDER;
+
+	private ApplicationEventPublisher publisher;
+
+	public TaskJobLauncherCommandLineRunner(JobLauncher jobLauncher,
+			JobExplorer jobExplorer) {
+		this.jobLauncher = jobLauncher;
+		this.jobExplorer = jobExplorer;
+	}
+
+	public void setOrder(int order) {
+		this.order = order;
+	}
+
+	@Override
+	public int getOrder() {
+		return this.order;
+	}
+
+	@Override
+	public void setApplicationEventPublisher(ApplicationEventPublisher publisher) {
+		this.publisher = publisher;
+	}
+
+	@Autowired(required = false)
```

Comment:
Given the `FactoryBean` for this class, I'd remove all `@Autowired` annotations from this class.